js-core/homeworks/homework-4/src/main.js

```diff
@@ -0,0 +1,183 @@
+
+/*
+* Напишите функцию которая будет принимать 1 аргумент и в зависимости от типа аргумента
+*
+* Если тип аргумента число или объект -> возвращать true
+* Если тип аргумента функция -> возвращать false
+* Если тип аргумента Строка и длина этой строки не равна 10 -> возвращать "длина вашей строки: <длина строки>
+*                            Если длина 10 -> 'you win'
+*
+*
+*
+*
+* */
+
+function returnValueFromTypeOfData (argum) {
+    
+    if(typeof argum === 'number' || typeof argum === 'object') {
+        return true;
+    }
+    
+    if(typeof argum === 'function') {
+        return false;
+    }
+    
+    if(argum.length !== 10 && typeof argum === 'string') {
+        return `Длина вашей строки: ${argum.length}`;
+    }
+    
+    if(argum.length === 10 && typeof argum === 'string') {
+        return `you win`;
+    }
+}
+
+console.log(returnValueFromTypeOfData(1));
+console.log(returnValueFromTypeOfData({}));
+console.log(returnValueFromTypeOfData(function someFunction () {}));
+console.log(returnValueFromTypeOfData(`12345678`));
+console.log(returnValueFromTypeOfData(`123456789`));
+console.log(returnValueFromTypeOfData(`1234567890`));
+
+
+/*
+*
+*  Задача в классе !
+*
+* */
+const webStore = {
+    catagory: {
+        noteBook: [`ASUS`, `Lenovo`, `Dell`, `HP`],
+        mobile: [`Nokia`, `Apple`, `Sony`],
+    }
+}
+
+const category = webStore.catagory;
+
+function catagoryFromWebStore (obj) {
+    
+    for(let key in category) {
+        
+        console.log(`Категория:`, key);
+        
+        for(let i = 0; i < category[key].length; i++) {
```

Comment:
it better to give a name to category[key] and than use it as a variable.
It would look much better